deployer/src/deployment/run.rs

```diff
@@ -14,44 +35,343 @@ pub async fn task(mut recv: RunReceiver, kill_send: KillSender) {
 
         let kill_recv = kill_send.subscribe();
 
+        let port = pick_unused_port().unwrap();
+        let addr = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), port);
+        let mut factory = abstract_factory.get_factory(ProjectName::from_str(&built.name).unwrap());
```

Comment:
What about if one of these worker tasks panics?

AFAICT, a pipeline is a collection of worker threads receiving their inputs and queuing their outputs by sending and receiving on mpsc channels. So if a worker thread panics for any reason, they drop their senders instances (if they have any) but since there are other instances running around (the ones in `Pipeline` itself e.g.), this won't close the channel, and thus won't kill `deployer`. So we'll have pipeline steps hanging on `recv` and nothing to actually send them.

It'd be preferable perhaps to unwind these panics (or generally run some clean up) when the worker dies (maybe via refactoring these workers into `struct`s and implementing `Drop` so that the channels get closed).